cmd/monogen/testapp/setup-helper.sh

```diff
@@ -6,8 +6,11 @@ set -e
 # Build the app.
 # Because we transitively depend on github.com/fjl/memsize, we need to disable checklinkname in go1.23.0 and higher.
 goVersion=$(go env GOVERSION)
-[[ $goVersion > go1.23.0 || $goVersion == go1.23.0 ]] && ldflags="-ldflags=-checklinkname=0"
-go build $ldflags -o testappd ./cmd/testappd
+if [[ $goVersion > go1.23.0 || $goVersion == go1.23.0 ]]; then
+    GOFLAGS="$GOFLAGS -ldflags=-checklinkname=0"
+fi
+export GOFLAGS
```

Comment:
Will this append `-ldflags=-checklinkname=0` to the user's exported `GOFLAGS` even if it's already present? If the script is run multiple times will it keep appending `-ldflags=-checklinkname=0`?

Reply:
Actually, we probably don't need to worry about these things.

Key idea: environment variables are copied into child processes.

> Will this append -ldflags=-checklinkname=0 to the user's exported GOFLAGS even if it's already present?

No, the script it will only modify the env var in its own process (not its parent's process).

> If the script is run multiple times will it keep appending -ldflags=-checklinkname=0?

No, since the script will take the `GOFLAGS` var from the parent process and only modify it within its own process.